### ISSUE TITLE
Error when filtering by relation

### ISSUE THREAD

**Issue by sajsanghvi** [neo4j-graphql/neo4j-graphql#107]
```
query{
  User(username:"Saj"){
    likes(filter:{user:{followers_some:{username:"Saj"}}}){
        user{
          username
        }
      }
  }
}
```

gives this error message

```
"Exception while fetching data (/User) : 

Invalid input 'Y': expected 'd/D' (line 4, column 147 (offset: 244))\n\"

[ (`User`)-[:`LIKES`]->(`User_likes`:`Like`) WHERE ( ALL(x IN [(`User_likes`)<-[:`LIKES`]-(`User_likes_User`) | ( `User_likes_User`.`followers` ANY {username:\"Saj\"})] WHERE x)) | `User_likes` {`_labels` : graphql.labels(`User_likes`), `user` : head([ (`User_likes`)<-[:`LIKES`]-(`User_likes_user`:`User`)  | `User_likes_user` {`_labels` : graphql.labels(`User_likes_user`), .`username`}])}] AS `likes`\"\n  
```

**Comment by sajsanghvi** [comment]
This is a problem that occurs whenever you filter by a relation within a relation in this example its Followers within a User within a Like

here is my Schema
```
type Like @model {
   user: User @relation (name:"LIKES", direction:IN)
   createdAt:Float
}
type User @model{
   followers: [User] @relation (name:"FOLLOWS", direction: IN)
   following: [User] @relation (name:"FOLLOWS", direction: OUT)
   username: String!
   createdAt:Float
   likes: [Like] @relation (name:"LIKES", direction: OUT)
}
```

**Comment by jexp** [comment]
can you try:

```
query{
  User(username:"Saj"){
    likes(filter:{followers_some:{username:"Saj"}}){
        user{
          username
        }
      }
  }
}
```

**Comment by jexp** [comment]
Should be fixed in 3.3.0.3 and 3.4.0.1 please check and let me know.

**Comment by sajsanghvi** [comment]
You are a god!
